torch/distributed/fsdp/shard_utils.py

```diff
@@ -0,0 +1,144 @@
+import bisect
+import itertools
+import math
+from typing import List, Tuple, Optional
+
+import torch
+from torch.distributed import ProcessGroup
+from torch.distributed._shard.sharded_tensor import ShardedTensor
+from torch.distributed._shard.sharding_spec import (
+    ChunkShardingSpec,
+    EnumerableShardingSpec,
+    ShardingSpec,
+)
+from torch.distributed.nn.functional import (
+    all_to_all_single,
```

Comment:
I'm not sure if we actually formally support comm. APIs in dist.nn.functional, as from previous discussions I remember they had/have a couple of issues (@cbalioglu @mrshenli probably know better). In general, the APIs in `distributed_c10d` are battle-tested and well-supported, so could we consider using a2a from there, unless there is a specific need to have autograd know about this a2a call?